### ISSUE TITLE
Support Bring Your Own Key for Object Storage

### ISSUE THREAD

**Issue by regevran** [scylladb/scylladb#22062]
Scylla should support user's own key for encrypting data at object storage.
Without this key Scylla does not encrypt the data and leave it for the object storage provider (S3 for example) to take care for encryption.
Refs: https://github.com/scylladb/scylla-enterprise/issues/4981

In general, the flow is always the same:
in the media: encrypted with a key for this media.
When transitioned - decryption with the key for this media and encryption with the keys for the transition to media.
For example:
On local disk - encrypted for local disk.
When streamed to S3: decryption with local disk keys and encryption with TLS.
If the user brings its own keys - before the TLS - encryption using the user's keys.
On S3 (not our responsibility, just to complete the picture) - decryption using TLS, encryption using S3's encryption.

**Comment by mykaul** [comment]
This is not realistic for 2025.1.0.
@regevran - please re-target.

**Comment by elcallio** [comment]
EAR is now properly applied to S3 storage (non-backup usage). Thus, in a sense at least, BYOK as applied to "normal" scylla EAR also applies here. 
BYOK to S3 bucket is wholly administered on AWS level.

**Comment by elcallio** [comment]
Not sure. I think what @regevran is at least partially referencing here is client side encryption of S3 data, outside of pre-S3 encryption (scylla EAR) and storage level managed encryption (S3 bucket ear). 
This we do not support. Not sure if it is anything we really care about either, outside maybe backup context.

**Comment by regevran** [comment]
Yes, @elcallio is right - the issue still cannot be closed and I am not sure we really want to implement it.

**Comment by regevran** [comment]
I need to take a deeper look into it. I prefer keeping the issue opened until we resolved the direction.
